### PR TITLE
feat: Local DID Provider APIs

### LINKED ISSUE #8: DID Provider APIs
Define DID Provider API's and provide basic implementation

### PR DESCRIPTION
**Pull request by rolsonquadras**
Define Local DID Provider API's and provide basic implementation

Closes #8

Signed-off-by: Rolson Quadras <rolson.quadras@securekey.com>

### CODE REVIEW

File: pkg/did/basic/did_basic_provider.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	"github.com/trustbloc/did-comm-go/pkg/did"
+)
+
+type provider struct {
+	store map[string]*did.LocalDIDInfo
+	lock  sync.RWMutex
+}
+
+// NewProvider instance of Basic DID provider
+func NewProvider() did.Provider {
```
Comment (troyronda) [inline pkg/did/basic/did_basic_provider.go]: Why not return *provider here?

Comment (troyronda) [reply pkg/did/basic/did_basic_provider.go]: (*Provider).

Comment (rolsonquadras) [reply pkg/did/basic/did_basic_provider.go]: It returns an interface.

Comment (rolsonquadras) [reply pkg/did/basic/did_basic_provider.go]: Removed interface return and updated to struct.

File: pkg/did/basic/did_basic_provider.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+// NewProvider instance of Basic DID provider
+func NewProvider() *Provider {
+	return &Provider{
+		store: map[string]*did.LocalDIDInfo{},
+	}
+}
+
+// CreateLocalDID create a new DID along with keypair and stores info along with metadata.
+func (prov Provider) CreateLocalDID(metadata map[string]interface{}) (*did.LocalDIDInfo, error) {
```
Comment (troyronda) [inline pkg/did/basic/did_basic_provider.go]: The receiver should be a pointer type

Comment (rolsonquadras) [reply pkg/did/basic/did_basic_provider.go]: @troyronda updated.

File: pkg/did/core/provider/basic/did_basic_provider.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+// NewProvider instance of Basic DID provider
+func NewProvider() *Provider {
+	return &Provider{
+		store: map[string]*didprovider.LocalDIDInfo{},
+	}
+}
+
+// CreateLocalDID create a new DID along with keypair and stores info along with metadata.
```
Comment (llorllale) [inline pkg/did/core/provider/basic/did_basic_provider.go]: @rolsonquadras what exactly is `metadata`?

Comment (rolsonquadras) [reply pkg/did/core/provider/basic/did_basic_provider.go]: Its a generic map to store metadata about the local DID. e.g, the recipient DIDs associated with this particular Local DID.

**Pull request merged**